lib/cli.rb

```diff
@@ -41,7 +41,7 @@ class Cli
   end
 
   post do |global_options,command,options,args|
-    if command.is_a?(GLI::Commands::Help) && !global_options[:version] && ARGV == ["help"]
+    if command.is_a?(GLI::Commands::Help) && !global_options[:version] || ARGV == ["help"]
```

Comment:
I think it would be good to add some parentheses here to make the logic more obvious.

And how about adding an integration test for that hint?